geode-core/src/main/java/org/apache/geode/internal/cache/ha/HARegionQueue.java

```diff
@@ -2711,7 +2711,7 @@ public void run() {
             boolean interrupted = Thread.interrupted();
             try {
               synchronized (this) {
-                this.wait(messageSyncInterval * 1000);
+                this.wait(1000 * (long) messageSyncInterval);
```

Comment:
Perhaps for increased readability this should be `(1000L * (long) messageSyncInterval)`. Even better would be to use `TimeUnit` for this conversion.

Reply:
TimeUnit at the moment has performance issues which were fixed as a part of JDK9 [https://bugs.java.com/bugdatabase/view_bug.do?bug_id=8152083]